### DISCUSSION TITLE
Need ADB command or Dynamic/Desktop Shortcuts

### DISCUSSION THREAD

**Discussion by Perpetual-Memories** [Nain57/Smart-AutoClicker#887] [General Feature Requests]
### Describe the feature

[Translate using Tongyi Qianwen]
Add external control capabilities, such as for the app [Extinguish](https://github.com/Moderpach/Extinguish), which allows controlling application services using shell commands.
Integrate specific scripts into the app's dynamic shortcuts or send them to desktop shortcuts, while keeping the status bar tiles to open the most recent scripts.

### Describe the usage

This would enable users to manage the app's state from their computer or integrate with apps like Tasker for automation purposes.
Shortcuts can provide quick access to scripts without needing to enter the app, offering convenience.

### Is it a duplicate?

- [X] There is no duplicates

**Comment by zhrmiah** [comment]
Yes I was thinking the same. Unfortunately macro is not easy to execute with any kind of quick trigger. Is it possible to integrate this with Tasker? Also if possible adding one click floating shortcut trigger button for macro can help different workflow execution faster.

**Reply by vibhor1102** [reply]
Not sure if you're still in requirement of the same, but I'm working on integration with Tasker by declaring a Locale plugin within the Klick'r app. Development should be completed this week, and it could be part of v4.1, possibly as soon as later this month (or next month).

I only have MacroDroid myself for testing, so by any chance could you beta-test the plugin for tasker as well to see if it works?

**Comment by Perpetual-Memories** [comment]
> Yes I was thinking the same. Unfortunately macro is not easy to execute with any kind of quick trigger. Is it possible to integrate this with Tasker? Also if possible adding one click floating shortcut trigger button for macro can help different workflow execution faster.

Thank you for your response.
Oh, yes, if it is difficult to quickly start macros using shell commands (by calling out the floating controller and automatically starting the execution), then having a floating controller to launch specified macros is indeed a good idea. This raises another issue: to start a specific macro rather than the most recently used one, macros need to be numbered. Therefore, if the floating controller to launch a certain macro could be pinned to the app's dynamic shortcuts, we could use the app Shortcut (Package name:net.ankio.shortcut) to obtain the shell command for using that dynamic shortcut, which would be much better. The remaining steps can be automated using an automation app by tapping coordinates to start the macro.
Additionally, I have replaced the app Tasker with another similar app called [ShortX](https://shortx-repo.github.io/ShortX-Pages/en/)—. Could you please get in touch with the developer of the app via [Telegram Group](https://t.me/shortxmod)?
Thank you again for your work.

**Reply by vibhor1102** [reply]
Not sure if you're still in requirement of the same, but I'm working on integration with Tasker by declaring a Locale plugin within the Klick'r app. Development should be completed this week, and it could be part of v4.1, possibly as soon as later this month (or next month).

I only have MacroDroid myself for testing, so by any chance could you beta-test the plugin for tasker as well to see if it works?

(Note, I've went through Locale Plugin route, not adb commands route. The plugin approach is more user friendly in general, and chosen for that convenience factor.)

**Comment by mruac** [comment]
Preferring either intent / adb / Tasker support for me.

Otherwise at the moment I am clicking on the view IDs of the scenario. As long as the scenario name is unique, the view ID will be also.

**Reply by vibhor1102** [reply]
Not sure if you're still in requirement of the same, but I'm working on integration with Tasker by declaring a Locale plugin within the Klick'r app. Development should be completed this week, and it could be part of v4.1, possibly as soon as later this month (or next month).

I only have MacroDroid myself for testing, so by any chance could you beta-test the plugin for tasker as well to see if it works?
